docs/api-docs/apps/tutorials/sample-app-next/05-step-3-integrate-api.md

```diff
@@ -0,0 +1,920 @@
+    apiVersion: 'v3'
+});
+
+const bigcommerceSigned = new BigCommerce({
+    secret: CLIENT_SECRET,
+    responseType: 'json'
+});
+```
+
+5. Remove the `QueryParams` interface.
+
+```js
+//Delete this code
+interface QueryParams {
+   [key: string]: string;
+}
+```
+
+6. Below the `bigcommerceSigned` variable, export the `bigcommerceClient` function.
+
+```js
+export function bigcommerceClient(accessToken: string, storeHash: string) {
+    return new BigCommerce({
+        clientId: CLIENT_ID,
+        accessToken,
+        storeHash,
+        responseType: 'json',
+        apiVersion: 'v3'
+    });
+}
+```
+
+7. Export `getBCAuth` and `getBCVerify` functions.
+
+```js
+export function getBCAuth(query: QueryParams) {
+    return bigcommerce.authorize(query);
+}
+
+export function getBCVerify({ signed_payload }: QueryParams) {
+    return bigcommerceSigned.verify(signed_payload);
+}
+``` 
+
+8. Add the `setSession`, `getSession`, and `removeSession` functions.
+
+```js
+export async function setSession(session: SessionProps) {
+    db.setUser(session);
+    db.setStore(session);
+}
+
+export async function getSession({ query: { context = '' } }: NextApiRequest) {
+    if (typeof context !== 'string') return;
+    const decodedContext = decodePayload(context)?.context;
+    const accessToken = await db.getStoreToken(decodedContext);
+
+    return { accessToken, storeHash: decodedContext };
+}
+
+export async function removeSession(res: NextApiResponse, session: SessionProps) {
+    await db.deleteStore(session);
+}
+```
+
+9. Add the `encodePayload` and `decodePayload` functions.
+
+```js
+export function encodePayload(context: string) {
+    return jwt.sign({ context }, JWT_KEY, { expiresIn: '24h' });
+}
+
+export function decodePayload(encodedContext: string) {
+    return jwt.verify(encodedContext, JWT_KEY);
+}
+```
+
+[View code in GitHub](https://github.com/bigcommerce/sample-app-nodejs/blob/step-3-add-database/lib/auth.ts)
+
+## Add a database
+
+In the purview of this tutorial, we will provide config and initialization code for both Firebase and MySQL databases. Depending on the database you choose to integrate your app with, use the configuration instructions specific to your setup. 
+
+Start by updating your environment variables.
+
+1. In the `.env` file, specify the database type.
+
+```
+DB_TYPE={database}
+```
+
+If using Firebase, add the following Firebase config placeholder:
+
+```
+FIRE_API_KEY={firebase key}
+FIRE_DOMAIN={firebase domain}
+FIRE_PROJECT_ID={firebase project id}
+```
+
+Jump to instructions on how to [set up your Firebase database](#set-up-the-firebase-database). 
+
+If using MySQL, add the following MySQL config placeholder:
+
+```
+MYSQL_HOST={mysql host}
+MYSQL_DATABASE={mysql domain}
+MYSQL_USERNAME={mysql username}
+MYSQL_PASSWORD={mysql password}
+MYSQL_PORT={mysql port}
+```
+Jump to instructions on how to [set up your MySQL database](#set-up-the-mysql-database). 
+
+## Set up the Firebase database
```

Comment:
I almost wonder if we should make the Firebase and mysql sections their own pages or if there's a way to make them collapsable? It might help to shorten step 3

Reply:
I don't think our current UI supports that but this could be an option once we move to our new UI. FYI, @uwikaiddi 
I'll see if I can extract the database setup steps without affecting the flow of the tutorial.

Reply:
@bc-chaz @uwikaiddi I don't see how we can make this work with the current UI. I could potentially remove the sections that cover how to create a Firbase project, Firebase config, and Firestore database since that information is available on their website. If we make Firebase and MySQL their own articles, the readers will have to navigate out of the tutorial to complete that setup. I don't think that will lead to a positive user experience. Alternatively, we can break step 3 into two sub-steps: one explaining how to integrate the BC API and another one covering the database piece. What do you think?